nodes.py

```diff
@@ -0,0 +1,76 @@
+from typing import Union
+from tokens import TokenType
+
+class Node:
+    def __init__(self) -> None:
+        pass
+
+    def __repr__(self) -> str:
+        pass
+
+    def eval(self): 
+        '''For actually evaluating the term, factor, or expression.'''
+        pass
+
+class BinaryOpNode(Node):
+    def __init__(self, type: TokenType) -> None:
+        super().__init__()
+        self.type: TokenType = type
+        self.left: Node = None
+        self.right: Node = None
+        if type == TokenType.PLUS:
+            self.func = lambda a,b : a+b 
+            self.symbol = '+'
+        elif type == TokenType.MINUS:
+            self.func = lambda a,b : a-b
+            self.symbol = '-'
+        elif type == TokenType.MULT:
+            self.func = lambda a,b : a*b
+            self.symbol = '*'
+        elif type == TokenType.DIV:
+            self.func = lambda a,b : a/b
```

Comment:
I noticed when doing 2 / 1, we get 2.0 instead of 2. Looks like Python handles '/' as true division whereas we should be using '//' for floor division.. A possible revision would check if we NEED to do one of the other.

`self.func = lambda a,b : a//b if a % b == 0 else a/b
`
`self.symbol = '/'`